dali/operators/generic/reduce/reduce.h

```diff
@@ -28,16 +28,53 @@
 #define REDUCE_TYPES (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t, float)  // NOLINT
 
 namespace dali {
+namespace detail {
+
+class AxesHelper {
+ public:
+  explicit inline AxesHelper(const OpSpec &spec) {
+    const bool has_axes_arg = spec.HasArgument("axes");
+    const bool has_axis_names_arg = spec.HasArgument("axis_names");
+
+    DALI_ENFORCE(!has_axes_arg || !has_axis_names_arg,
+      "Arguments `axes` and `axis_names` are mutually exclusive");
+
+    if (has_axis_names_arg || !has_axes_arg) {
+      axis_names_ = spec.GetArgument<TensorLayout>("axis_names");
+      axes_ = {};
+    } else {
+      axes_ = spec.GetRepeatedArgument<int>("axes");
+      axis_names_ = TensorLayout{};
+    }
+  }
+
+  void PrepareAxes(TensorLayout &&layout, int sample_dim) {
+    if (!axis_names_.empty()) {
+      axes_ = GetDimIndices(layout, axis_names_).to_vector();
+    }
+
+    if (axes_.size() == 0) {
```

Comment:
Don't rely on size() - it's impossible to distinguish "axes=[]" from "axes=None" - we want the latter to mean "all axes", but no the former. Use a boolean flag to check that no axis indices were provided.

Reply:
Done